lib/ui/common/ActionSheet.dart

```diff
@@ -0,0 +1,146 @@
+
+  ActionSheetStyle(
+      {this.backgroundColor,
+      this.indicatorLineColor,
+      this.confirmButtonBackgroundColor,
+      this.cancelButtonBackgroundColor,
+      this.indicatorLineEdgePadding,
+      this.cornerRadius,
+      this.indicatorLineRadius,
+      this.indicatorLineThickness,
+      this.indicatorLineWidth});
+
+  factory ActionSheetStyle.defaultStyle() {
+    return ActionSheetStyle(
+      backgroundColor: const Color(0xFFffffff),
+      indicatorLineColor: const Color(0xFFe0e1ee),
+      confirmButtonBackgroundColor: const Color(0xFF24d900),
+      cancelButtonBackgroundColor: const Color(0xFFe9eaf2),
+      indicatorLineEdgePadding: const EdgeInsets.only(top: 8, bottom: 50),
+      cornerRadius: const Radius.circular(40),
+      indicatorLineRadius: const Radius.circular(2.5),
+      indicatorLineThickness: 5,
+      indicatorLineWidth: 40,
+    );
+  }
+
+  ActionSheetStyle copyWith(
+      {Color backgroundColor,
+      Color indicatorLineColor,
+      Color confirmButtonBackgroundColor,
+      Color cancelButtonBackgroundColor,
+      EdgeInsets indicatorLineEdgePadding,
+      Radius cornerRadius,
+      Radius indicatorLineRadius,
+      double indicatorLineThickness,
+      double indicatorLineWidth}) {
+    return ActionSheetStyle(
+      backgroundColor: backgroundColor ?? this.backgroundColor,
+      indicatorLineColor: indicatorLineColor ?? this.indicatorLineColor,
+      confirmButtonBackgroundColor:
+          confirmButtonBackgroundColor ?? this.confirmButtonBackgroundColor,
+      cancelButtonBackgroundColor:
+          cancelButtonBackgroundColor ?? this.cancelButtonBackgroundColor,
+      indicatorLineEdgePadding:
+          indicatorLineEdgePadding ?? this.indicatorLineEdgePadding,
+      cornerRadius: cornerRadius ?? this.cornerRadius,
+      indicatorLineRadius: indicatorLineRadius ?? this.indicatorLineRadius,
+      indicatorLineThickness:
+          indicatorLineThickness ?? this.indicatorLineThickness,
+      indicatorLineWidth: indicatorLineWidth ?? this.indicatorLineWidth,
+    );
+  }
+}
+
+class ActionSheet extends StatelessWidget {
+  final ActionSheetViewModel _viewModel;
+  final ActionSheetStyle _style;
+  final ActionSheetListItemStyle _cellStyle;
+
+  const ActionSheet(
+      {Key key,
+      ActionSheetViewModel viewModel,
+      ActionSheetStyle style,
+      ActionSheetListItemStyle cellStyle})
+      : this._viewModel = viewModel,
+        this._style = style,
+        this._cellStyle = cellStyle,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+        color: _style.cornersColor,
+        child: Container(
+            decoration: BoxDecoration(
+                color: _style.backgroundColor,
+                borderRadius: BorderRadius.only(
+                    topLeft: _style.cornerRadius,
+                    topRight: _style.cornerRadius)),
+            child: HeaderAndFooterListView.builder(
```

Comment:
The header and footer widget is so that items are dynamically created (so we can have infinite lists), this only works if the header and footer widget is not nested in other components, so there's no point in using it. The action list will never be more than 4 or 5 items so this is not needed. Lets change to a standard static list.

Reply:
Now it's changed as a Column